packages/verify-env/src/index.test.ts

```diff
@@ -0,0 +1,37 @@
+import { it, expect, vi } from "vitest";
+import { extractEnvVars, verify, validateVariables } from "./verify.ts";
+import fs from "fs";
+
+it("extracts env vars from a string", () => {
+  const source = fs.readFileSync(__dirname + "/testSource.txt").toString();
+  const envVars = extractEnvVars(source);
+  expect(envVars).toStrictEqual([
+    "REACT_APP_ENV",
+    "REACT_APP_INFURA_ID",
+    "REACT_APP_WALLETCONNECT_PROJECT_ID",
+  ]);
+});
+
+it("verifies vars from files", () => {
+  const consoleMock = vi
+    .spyOn(console, "error")
+    .mockImplementation(() => undefined);
+  const processMock = vi.spyOn(process, "exit").mockImplementation(() => {
+    return undefined as never;
+  });
+
+  verify(__dirname + "/test");
+  expect(processMock).toHaveBeenCalledOnce();
+  expect(processMock).toHaveBeenCalledWith(1);
```

Comment:
Isn't there any canonical way for plugins to signal non-recoverable errors and let webpack exit? Calling `process.exit` from inside a plugin seems very unplugin-y.

Reply:
there seems to be a way, but in the interest of making this plugin bail as soon as possible (not wait for others to finish), i'm gonna leave it as this for now.